torch/csrc/jit/backends/backend.h

```diff
@@ -0,0 +1,155 @@
+std::function<void(Stack&)> getPreprocessFunc() {
+  return [](Stack& stack) {
+    auto method_compile_spec = pop(stack).toGenericDict();
+    auto mod = pop(stack);
+    auto self = pop(stack).toCustomClass<TBackendInterface>();
+    auto ret = self->preprocess(mod, method_compile_spec);
+    push(stack, ret);
+  };
+}
+
+c10::FunctionSchema getCompileSchema() {
+  c10::Argument self("self", c10::AnyType::get());
+  c10::Argument mod("processed", c10::AnyType::get());
+  auto any_dict_ty =
+      c10::DictType::create(c10::StringType::get(), c10::AnyType::get());
+  c10::Argument method_compile_spec("method_compile_spec", any_dict_ty);
+  c10::Argument handles("handles", any_dict_ty);
+
+  c10::FunctionSchema compile_schema(
+      "compile",
+      /*overload_name=*/"",
+      /*arguments=*/{self, mod, method_compile_spec},
+      /*returns=*/{handles});
+  return compile_schema;
+}
+
+template <typename TBackendInterface>
+std::function<void(Stack&)> getCompileFunc() {
+  return [](Stack& stack) {
+    auto method_compile_spec = pop(stack).toGenericDict();
+    auto processed = pop(stack);
+    auto self = pop(stack).toCustomClass<TBackendInterface>();
+    auto ret = self->compile(processed, method_compile_spec);
+    push(stack, ret);
+  };
+}
+
+c10::FunctionSchema getExecuteSchema() {
+  c10::Argument mod("processed", c10::AnyType::get());
+  // TODO: This is actually not accurate because execute is varret, but you can
+  // only use a FunctionSchema with one return type to create a
+  // jit::BuiltinOpFunction (which is what) torch::class_ does under the hood.
+  return c10::FunctionSchema(
+      "execute",
+      /*overload_name=*/"",
+      /*arguments=*/{},
+      /*returns=*/{mod},
+      true,
+      true);
+}
+
+template <typename TBackendInterface>
+std::function<void(Stack&)> getExecuteFunc() {
+  return [](Stack& stack) {
+    // TODO: Stack does not have the number of arguments on it (probably because
+    // schema is not truly varret) so the best we can do is to assume that
+    // everything on the stack is an argument to execute.
+    auto num_args = stack.size();
+    auto args = pop(stack, num_args);
+    TORCH_INTERNAL_ASSERT(args.size() >= 1);
+
+    auto self = args[0].toCustomClass<TBackendInterface>();
+    auto methodStack = pop(args, args.size() - 1);
+    // FIXME: We can't actually let `execute` operate on the stack in-place,
+    // since we need to pop `self` off anyway. So we can make Backend::execute
+    // more straightforward.
+    self->execute(methodStack);
+
+    // TODO: Since there is only one return value as per the schema, sneak in
+    // multiple return values if needed by packing them into a Tuple. Fix this
+    // so that multiple arguments can be passed back as is without intervention.
+    if (methodStack.size() == 1) {
+      push(stack, std::move(methodStack[0]));
+    } else {
+      push(stack, c10::ivalue::Tuple::create(methodStack));
+    }
+  };
+}
```

Comment:
Are these five functions expected to be user-visible? if not, can we put them in a [detail namespace](https://github.com/pytorch/pytorch/blob/06928047478fab33678bd9248886a938b9b07937/torch/custom_class_detail.h#L10) and potentially in a different file? I just ask because I was reading this header as if I were a new user and I was confused as to what's an external API v.s. what's an implementation